docs/api/session.md

```diff
@@ -1360,6 +1360,36 @@ specified when registering the protocol.
 
 Returns `Promise<void>` - resolves when the code cache clear operation is complete.
 
+#### `ses.getSharedDictionaryUsageInfo()`
+
+Returns `Promise<SharedDictionaryUsageInfo[]>` - an array of shared dictionary information entries in Chromium's networking service's storage.
+
+Shared dictionaries are used to power advanced compression of data sent over the wire, specifically with Brotli and ZStandard. You don't need to call any of the shared dictionary APIs in Electron to make use of this advanced web feature, but if you do, they allow deeper control and inspection of the shared dictionaries used during decompression.
+
+To get detailed information about a specific shared dictionary entry, call `getSharedDictionaryInfo(options)`.
+
+#### `ses.getSharedDictionaryInfo(options)`
+
+* `options` Object
+  * `frameOrigin` string - The origin of the frame where the request originates. It’s specific to the individual frame making the request and is defined by its scheme, host, and port. In practice, will look like a URL.
+  * `topFrameSite` string - The site of the top-level browsing context (the main frame or tab that contains the request). It’s less granular than `frameOrigin` and focuses on the broader "site" scope. In practice, will look like a URL.
+
+Returns `Promise<SharedDictionaryInfo[]>` - an array of shared dictionary information entries in Chromium's networking service's storage.
+
+To get information about all present shared dictionaries, call `getSharedDictionaryUsageInfo()`.
+
+#### `ses.clearSharedDictionaryCache()`
+
+Returns `Promise<void>` - resolves when the dictionary cache has been cleared, both in memory and on disk.
+
+#### `ses.clearSharedDictionaryCacheForIsolationKey(options)`
```

Comment:
`session.clearData` calls an internal class `content::BrowsingDataRemover`. That class has support for embedders defining their own data to remove via [content::BrowsingDataRemoverDelegate::RemoveEmbedderData](https://source.chromium.org/chromium/chromium/src/+/main:content/public/browser/browsing_data_remover_delegate.h;l=56-65;drc=c325fff94cab7b3d378b5f367f75249662bbeaeb).

Do you think removing the data as part of our embedder data would be a good fit? It would have potential to remove the need for this extra method.

Reply:
Appreciate the call out! Chatted with Sam about this and he made a good point that the shared dictionaries are Chrome's data and not Electron's data as the embedders. There is intentionally a seperation with an designated API to clear the dictionary cache. Also I'm not sure the the removeEmbedderData can support us clearing the dictionary for a specific key!

Reply:
Looking deeper into this, the BrowsingDataRemover actually handles clearing the SharedDictionary cache here:
https://source.chromium.org/chromium/chromium/src/+/main:content/browser/browsing_data/browsing_data_remover_impl.cc;l=663-675;drc=277f4ab48eb85f7441f78aed191c31068ce89814

Using this snippet should clear it for a specific origin
```
session.clearData({ dataTypes: ['cache'], origin: 'https://www.electronjs.org' })
```

Will this accomplish the same needs as `clearSharedDictionaryCacheForIsolationKey`?

Reply:
There is a valid developer use case for clearing the dictionaries and not the entire cache. That's why chrome has this debugging page. I'd prefer to retain this individual API, perhaps documenting the clearing cache also wipes these.

Reply:
Makes sense! I can add a small bit to the `session.clearData` docs to mention:

Note: Clearing `cache` data  will also clear the shared dictionary cache, which may affect network-related optimizations like compression dictionaries used in network traffic (if enabled). This means that any dictionaries used for compression may be reloaded after clearing the cache.
If you wish to clear the shared dictionary cache but leave other cached data intact, you may want to use  `ses.clearSharedDictionaryCache()`.